client/classic/upe/deferred-intent.js

```diff
@@ -42,14 +42,20 @@ jQuery( function ( $ ) {
 		}
 	}
 
-	// Mount the Stripe Payment Elements onto the add payment method page.
+	// Mount the Stripe Payment Elements onto the Add Payment Method page and Pay for Order page..
 	if (
 		$( 'form#add_payment_method' ).length ||
 		$( 'form#order_review' ).length
 	) {
 		maybeMountStripePaymentElement();
+
+		// This function runs before WooCommerce has attached its callbacks in tokenization-form.js so we need to add a slight delay and trigger the event again.
+		setTimeout( () => {
```

Comment:
I had to add this dirty fix to fix an issue where our payment elements were mounted before WC Core could attach its callbacks in tokenization-form.js which handle the payment method radio buttons.

This means that without this timeout, the payment elements were loaded being displayed and they couldn't be hidden by checking or unchecking the radio buttons. 


For example:

https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/05296cbe-f7bb-49b7-904a-6d2a3daeab68

Adding this trigger to run on a very tiny delay fixes it. If you have another idea for how to make sure we attach afterwards, let me know

Reply:
I noticed WooPayments [enqueues its scripts on `wp_footer`](https://github.com/Automattic/woocommerce-payments/blob/cc429402e17b3d20347915d181ebaf42a9bb2945/includes/class-wc-payments-checkout.php#L194) so maybe this is how they're getting around similar issues?

I just removed this setTimeout and reproduced the issue, then I replaced the `add_action( 'wp_enqueue_scripts', [ $this, 'payment_scripts' ] );` found [here in the `WC_Stripe_UPE_Payment_Gateway` class](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/e2d607b872e7a8a42f0fd2d17baf78d6b204dfc2/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L157) and that seems to fix it.

Reply:
Ah thanks for looking into how WooPayments handles this. I've submitted those changes in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2801/commits/f3015688dec8abc294dd2355bd21ab9421ea946b